### PR TITLE
vnext regen resources, fix api-version method parameter

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
fix https://github.com/Azure/azure-libraries-for-java/issues/1084

### CODE REVIEW

File: azure-mgmt-resources/src/main/java/com/azure/management/resources/AliasPathType.java [hunk 1]
```diff
@@ -1,7 +1,5 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for
-// license information.
-// 
+// Licensed under the MIT License.
```
Comment (xccc-msft) [inline azure-mgmt-resources/src/main/java/com/azure/management/resources/AliasPathType.java]: Should we apply this change to all existing files?

Comment (weidongxu-microsoft) [reply azure-mgmt-resources/src/main/java/com/azure/management/resources/AliasPathType.java]: We can get some script to do this. Maybe later when someone finished more important tasks...

Currently newly generated code will use this short style.

**Comment by weidongxu-microsoft** [comment]
PR on spec https://github.com/Azure/azure-rest-api-specs/pull/8803

**Comment by weidongxu-microsoft** [comment]
Fix https://github.com/Azure/azure-libraries-for-java/issues/1084

File: azure-mgmt-resources/src/main/java/com/azure/management/resources/DeploymentProperties.java [hunk 2]
```diff
@@ -24,7 +22,8 @@
     private Object template;
 
     /*
-     * Entity representing the reference to the template.
+     * The URI of the template. Use either the templateLink property or the
```
Comment (yungezz) [inline azure-mgmt-resources/src/main/java/com/azure/management/resources/DeploymentProperties.java]: is this change caused by generator change also?

File: azure-mgmt-resources/src/main/java/com/azure/management/resources/implementation/DeploymentsImpl.java [hunk 3]
```diff
@@ -40,8 +41,7 @@
 
     @Override
     public PagedIterable<Deployment> listByResourceGroup(String groupName) {
-        // FIXME: filter & top parameter
-        return this.manager().inner().deployments().listByResourceGroup(groupName, null, null).mapPage(inner -> createFluentModel(inner));
+        return this.manager().inner().deployments().listByResourceGroup(groupName).mapPage(inner -> createFluentModel(inner));
```
Comment (yungezz) [inline azure-mgmt-resources/src/main/java/com/azure/management/resources/implementation/DeploymentsImpl.java]: just confirm, this is change from generator

File: azure-mgmt-resources/src/main/java/com/azure/management/resources/DeploymentProperties.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply azure-mgmt-resources/src/main/java/com/azure/management/resources/DeploymentProperties.java]: Yes. I think it is from changes by modelerfour. Not confirmed.

File: azure-mgmt-resources/src/main/java/com/azure/management/resources/implementation/DeploymentsImpl.java (hunk 3 above)
Comment (weidongxu-microsoft) [reply azure-mgmt-resources/src/main/java/com/azure/management/resources/implementation/DeploymentsImpl.java]: Actually this is Impl change.

The resources RP is among the 1st to migrate. Hence there remains quite some TODO/FIXME.

The method call is actually same, but older generate does not generate the method with default parameters (hence the 2 `null`s in code before).

**Pull request merged**